From 93dc575eab5924bdcb54e9f8323d1da982572899 Mon Sep 17 00:00:00 2001
From: Nadav Hugim <nhugim@habana.ai>
Date: Mon, 20 Dec 2021 13:45:27 +0200
Subject: [PATCH] [SW-52631] boot_info: write ver to boot_info

send version from preboot to uboot using boot_info

Change-Id: Icb6f6aac6425112428727cdb75c4381cc85fdbab
Signed-off-by: Nadav Hugim <nhugim@habana.ai>
(cherry picked from commit 7b25e8bb36511b43bacf93ffb89e255940408005)
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index c01add1..bb08adc 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -160,6 +160,7 @@
 
 static void preboot_version(void)
 {
+	int err;
 	u32 btlver = regmap_readl(mmPREBOOT_VER_OFFSET);
 	unsigned long offset = (unsigned long) preboot_vers[PREBOOT_BTL_VER];
 	int i = 0, btl_ver_sz = 0;
@@ -183,6 +184,11 @@
 	}
 
 	regmap_writel((u32)(offset & ~SDRAM_BASE), mmPREBOOT_VER_OFFSET);
+
+	err = boot_info_set(HL_BOOT_INFO_PREBOOT_BTL_VERSION, preboot_vers[PREBOOT_BTL_VER],
+			VERSION_MAX_LEN);
+	if (err)
+		preboot_puts("Failed to pass preboot BTL version securely\n");
 }
 
 #define FUSE_VER_NOT_IMPLEMENT	"not implemented"
@@ -256,6 +262,11 @@
 	preboot_puts("\n");
 
 	regmap_writel((u32)(offset & ~SDRAM_BASE), mmFUSE_VER_OFFSET);
+
+	err = boot_info_set(HL_BOOT_INFO_PREBOOT_EFUSE_VERSION, preboot_vers[PREBOOT_EFUSE_VER],
+			VERSION_MAX_LEN);
+	if (err)
+		preboot_puts("Failed to pass EFUSE version securely\n");
 }
 
 __weak void preboot_security_init(void) {}
